tests/integration/panic.rs

```diff
@@ -0,0 +1,160 @@
+use chalk_integration::db::ChalkDatabase;
+use chalk_integration::query::LoweringDatabase;
+use chalk_ir::interner::ChalkIr;
+use chalk_ir::AssocTypeId;
+use chalk_ir::Goal;
+use chalk_ir::ImplId;
+use chalk_ir::InEnvironment;
+use chalk_ir::OpaqueTyId;
+use chalk_ir::Parameter;
+use chalk_ir::ProgramClause;
+use chalk_ir::StructId;
+use chalk_ir::TraitId;
+use chalk_ir::TypeName;
+use chalk_ir::UCanonical;
+use chalk_rust_ir::AssociatedTyDatum;
+use chalk_rust_ir::AssociatedTyValue;
+use chalk_rust_ir::AssociatedTyValueId;
+use chalk_rust_ir::ImplDatum;
+use chalk_rust_ir::OpaqueTyDatum;
+use chalk_rust_ir::StructDatum;
+use chalk_rust_ir::TraitDatum;
+use chalk_rust_ir::WellKnownTrait;
+use chalk_solve::RustIrDatabase;
+use chalk_solve::Solution;
+use chalk_solve::SolverChoice;
+use std::sync::Arc;
+
+#[derive(Debug, Default)]
+struct MockDatabase {
+    chalk_db: ChalkDatabase,
+    panic: bool,
+}
+
+impl MockDatabase {
+    pub fn with(program_text: &str, solver_choice: SolverChoice) -> Self {
+        Self {
+            chalk_db: ChalkDatabase::with(program_text, solver_choice),
+            panic: false,
+        }
+    }
+
+    pub fn solve(
+        &self,
+        goal: &UCanonical<InEnvironment<Goal<ChalkIr>>>,
+    ) -> Option<Solution<ChalkIr>> {
+        let solver = self.chalk_db.solver();
+        let solution = solver.lock().unwrap().solve(self, goal);
+        solution
+    }
+}
+
+impl RustIrDatabase<ChalkIr> for MockDatabase {
```

Comment:
For these methods, I was thinking just hardcode what we want to return for `trait Foo` and `struct Bar`